File: Napakalaki/src/napakalaki/Treasure.java
/*
* PDOO Practicas - Napakalaki Java
* @author Emilio Reguero y Dunia Rodríguez 
*/
package napakalaki;

public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind type;
    
    
    public Treasure(String n, int bonus, TreasureKind t){
        this.name = n;
        this.bonus = bonus;
        this.type= t;
    }
    
    String getName(){
        return name;
    }
    
    int getBonus(){
        return bonus;
    }
    
    TreasureKind getType(){
        return type;
    }
}

File: NapakalakiRuby/pkg/NapakalakiRuby-0.0.1/lib/treasure.rb
#encoding: utf-8
#Autor: Dunia Rodríguez y Emilio Reguero

require_relative "treasure_kind.rb"

module NapakalakiGame
class Treasure
  attr_reader :name, :bonus, :type
  
  def initialize(name, bonus, type)
    @name = name
    @bonus = bonus
    @type = type
  end
  
  def getType
   @type
  end 
  
  def to_s
    "Nombre: #{@name}, Bonus: #{@bonus}, Tipo: #{type.to_s}"
  end
end
end
